trace/filters.go

```diff
@@ -40,6 +42,39 @@ func NewHTTPHostFilterCollector(matchers []*regexp.Regexp, col Collector) Collec
 	}
 }
 
+// Allows only matching paths
+// TODO: compile the N regular expressions into one for efficiency.
+func NewHTTPPathAllowlistCollector(matchers []*regexp.Regexp, col Collector) Collector {
+	return &genericRequestFilter{
+		Collector: col,
+		filterFunc: func(r akinet.HTTPRequest) bool {
+			if r.URL != nil {
+				for _, m := range matchers {
+					if m.MatchString(r.URL.Path) {
+						return true
+					}
+				}
+			}
+			return false
+		},
+	}
+}
+
+// Allows only matching hosts
+func NewHTTPHostAllowlistCollector(matchers []*regexp.Regexp, col Collector) Collector {
+	return &genericRequestFilter{
+		Collector: col,
+		filterFunc: func(r akinet.HTTPRequest) bool {
+			for _, m := range matchers {
+				if m.MatchString(r.Host) {
+					return true
+				}
+			}
+			return false
+		},
+	}
+}
```

Comment:
Seems a shame that we can't just build the corresponding deny filter and invert it. Another time, perhaps.